app-frontend/src/app/components/lab/diagramNodeHeader/diagramNodeHeader.html

```diff
@@ -28,8 +28,36 @@
             ng-if="$ctrl.node.apply"></span>
       {{$ctrl.typeMap[$ctrl.node.type]}}
     </div>
-    <div class="node-header-label">
-      {{$ctrl.node.metadata.label}}
+    <div>
+      <span ng-if="!$ctrl.isEditNodeLabel">
+        <button class="btn node-button pate-title-edit detail-title-edit"
+                type="button"
+                ng-click="$ctrl.toggleNodeLabelEdit()">
+          <span class="icon-pencil"></span>
+        </button>
+        <span class="node-header-label">{{$ctrl.node.metadata.label}}</span>
+      </span>
+      <span ng-if="$ctrl.isEditNodeLabel">
+        <form class="inline-form" ng-submit="$ctrl.finishNodelabelEdit()">
+          <div class="form-group all-in-one">
```

Comment:
The default size of this input and the buttons aren't a great match for the node header. 

You'll want to add a class here or on the elements themselves that:
- removes all vertical padding from the input (it's the only way it will fit in the header)
- shrinks the buttons so that they don't force the size of the input to grow and so that they also fit in the header.
- adjusts the `z-index` of the inputs and buttons so that they dont get displayed under the other node controls